src/solid/Virtualizer.tsx

```diff
@@ -164,12 +174,20 @@ export const Virtualizer = <T,>(props: VirtualizerProps<T>): JSX.Element => {
     itemSize,
     overscan,
     undefined,
-    undefined,
+    cache,
     !itemSize
   );
   const resizer = createResizer(store, horizontal);
   const scroller = createScroller(store, horizontal);
 
+  // The elements length and cached items length are different just after element is added/removed.
+  if (props.data.length !== store.$getItemsLength()) {
+    store.$update(ACTION_ITEMS_LENGTH_CHANGE, [props.data.length, props.shift]);
+  }
+  if (props.startMargin !== store.$getStartSpacerSize()) {
+    store.$update(ACTION_START_OFFSET_CHANGE, props.startMargin ?? 0);
+  }
+
```

Comment:
This will not work, and implementation already exists:
```suggestion
```

https://github.com/inokawa/virtua/blob/4ff43ff0061bd5f6c48bae0888631fa35b25887a/src/solid/Virtualizer.tsx#L239-L259